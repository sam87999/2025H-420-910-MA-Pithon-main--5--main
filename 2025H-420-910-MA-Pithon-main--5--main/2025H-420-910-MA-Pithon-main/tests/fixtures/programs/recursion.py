# Test de Alexandrine Dub√©
def factorial(n):
    # Base case: The condition that stops the recursion
    if n == 0 or n == 1:
        return 1
    # Recursive case: The function calls itself with a modified argument
    else:
        to_send = n - 1
        new_val = factorial(to_send)
        mine = n * new_val
        return mine


result = factorial(5)
print(result) # Output: 120
print(factorial(10)) # Output: 3628800
print(factorial(34)) # Output: 295232799039604140847618609643520000000
print(factorial(0)) # Output: 1