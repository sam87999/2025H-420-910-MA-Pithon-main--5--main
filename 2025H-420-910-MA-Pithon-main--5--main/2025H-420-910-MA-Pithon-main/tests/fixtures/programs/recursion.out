120
3628800
295232799039604140847618609643520000000
1
